laboratory-service/src/main/java/com/raf/si/laboratoryservice/mapper/OrderMapper.java

```diff
@@ -0,0 +1,134 @@
+            String errMessage = String.format("Ne postoji radni nalog sa datim id-em: %s",
+                    saveRequest.getOrderId());
+            log.info(errMessage);
+            throw new BadRequestException(errMessage);
+        }
+
+        LabWorkOrder order = orderOptional.get();
+        if (order.getStatus().equals(OrderStatus.NEOBRADJEN)){
+            order.setStatus(OrderStatus.U_OBRADI);
+            labWorkOrderRepository.save(order);
+        }
+
+        Optional<AnalysisParameter> analysisParameterOptional =  analysisParameterRepository.findById(
+                saveRequest.getParameterId());
+
+        if(analysisParameterOptional.isEmpty()){
+            String errMessage = String.format("Ne postoji parametar sa datim id-em: %s",
+                    saveRequest.getOrderId());
+            log.info(errMessage);
+            throw new BadRequestException(errMessage);
+        }
+
+        AnalysisParameter analysisParameter = analysisParameterOptional.get();
+
+        Optional<AnalysisParameterResult> resultOptional = analysisParameterResultRepository
+                .findAnalysisParameterResultByLabWorkOrderAndAnalysisParameter(
+                        order, analysisParameter
+                );
+
+        if(resultOptional.isEmpty()){
+            String errMessage = "Ne postoji rezultat sa date parametre.";
+            log.info(errMessage);
+            throw new BadRequestException(errMessage);
+        }
+
+        AnalysisParameterResult result = resultOptional.get();
+        result.setResult(saveRequest.getResult());
+        result.setDateAndTime(new Date(System.currentTimeMillis()));
+        result.setLbzBiochemist(TokenPayloadUtil.getTokenPayload().getLbz());
+
+        analysisParameterResultRepository.save(result);
+    }
+
+    public OrderHistoryResponse orderPageToOrderHistoryForLabResponse(Page<LabWorkOrder> orderPage) {
+        List<OrderResponse> orders = orderPage.getContent()
+                .stream()
+                .filter(this::shouldRemove)
+                .map(this::orderToOrderResponse)
+                .collect(Collectors.toList());
+
+        return new OrderHistoryResponse(orders, orderPage.getTotalElements());
+    }
+
+    private boolean shouldRemove(LabWorkOrder order){
+        UUID pbo = TokenPayloadUtil.getTokenPayload().getPbo();
+        return !order.getReferral().getPboReferredTo().equals(pbo);
+    }
+
+    public ResultResponse orderToResultResponse(LabWorkOrder order){
+        ResultResponse response = new ResultResponse();
+        response.setOrder(order);
+        response.setParameterList(new ArrayList<>());
+        response.setAnalysisList(new ArrayList<>());
+        List<AnalysisParameterResult> results = order.getAnalysisParameterResults();
+        for(AnalysisParameterResult apr : results){
+            response.getAnalysisList().add(apr.getAnalysisParameter().getAnalysis());
+            response.getParameterList().add(apr.getAnalysisParameter());
```

Comment:
sa dodavanjem AnalysisParameter klase, dodajes i analizu i parametar vezan za tu tabelu. Ako je to potrebno super, ali ako nije treba napraviti dto koji ce uhvatiti samo informacije koje su potrebne